docs/fundamentals/connection/connect-to-mongodb.txt

```diff
@@ -0,0 +1,255 @@
+           // ...
+       ],
+     // ...
+   ]
+
+.. tip::
+
+   To retrieve your Atlas connection string, follow the
+   :ref:`Create a Connection String <laravel-quick-start-connection-string>`
+   step of the Quick Start tutorial.
+
+.. _laravel-other-ways-to-connect:
+
+Other Ways to Connect to MongoDB
+--------------------------------
+
+The following sections show you how to connect to a single MongoDB server
+instance or a replica set not hosted on MongoDB Atlas.
+
+.. _laravel-connect-localhost:
+
+Connect to a MongoDB Server on Your Local Machine
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This section shows an example connection string you can use when running a
+Laravel application and MongoDB server from the same machine, such as your
+local development environment.
+
+To connect your application to a MongoDB instance hosted on the same machine,
+you must complete the following tasks:
+
+- Download, install, and run MongoDB server.
+- Obtain the IP address and port on which your MongoDB server is running. If
+  you use the default settings of a local installation of MongoDB server,
+  the IP address is ``127.0.0.1``, and the port is ``27017``.
+- Set up your ``config/database.php`` connection to reference the environment
+  variable ``DB_URI`` for the value of the ``dsn``, as shown in the
+  :ref:`laravel-atlas-connection-example` section.
+
+The following example shows a sample connection string that you can add to the
+``.env`` file if your application connects to a MongoDB server running on the
+default IP address and port:
+
+.. code-block:: php
+   :caption: Sample .env environment configuration to connect to a local MongoDB server.
+
+   DB_URI="mongodb://127.0.0.1:27017/";
+
+To learn how to download  and install MongoDB server, see
+:manual:`Install MongoDB Community Edition </installation/#mongodb-installation-tutorials>`
+in the {+server-docs-name+}.
+
+.. _laravel-connect-replica-set:
+
+Connect to a Replica Set
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+A MongoDB replica set deployment is a group of connected instances, or nodes,
+where the nodes store the same data set. This configuration of instances
+provides data redundancy and high data availability.
+
+To connect to a replica set deployment, specify each node's hostname and port
+number, separated by commas, and the replica set name as the value of the
+``replicaSet`` parameter in the connection string.
+
+This example, which shows the connection string you can add to your
+Laravel application's ``.env`` file to connect to a replica set, uses the
+following sample values:
+
+- ``host1``, ``host2``, and ``host3`` as the hostnames of the MongoDB nodes
+- ``27017`` as the port on which MongoDB runs on those hosts
+- ``myRS`` as the configured name of the replica set
+- ``myUser`` and ``myPass123`` as the credentials of a database user
+
+.. code-block:: bash
+
+   DB_URI="mongodb://myUser:myPass123@host1:27017,host2:27017,host3:27017/?replicaSet=myRS"
```

Comment:
After testing on a local replica set, I can confirm that the `replicatSet` option works.
```env
MONGODB_URI="mongodb://localhost:27017,localhost:27018,localhost:27019/"
```
```php
        'mongodb' => [
            'driver' => 'mongodb',
            'dsn' => env('MONGODB_URI'),
            'database' => 'laravel',
            'options' => [
                'replicaSet' => 'myRS',
            ]
        ]
```

You can also pass the list of hosts like this:
```php
        'mongodb' => [
            'driver' => 'mongodb',
            'host' => [
                '127.0.0.1:27017',
                '127.0.0.1:27018',
                '127.0.0.1:27019',
            ],
            'database' => 'laravel',
            'options' => [
                'replicaSet' => 'myRS',
            ]
        ]
```